Automattic/wp-calypso
# Pattern Assembler: Add the activation screen

Related to https://github.com/Automattic/wp-calypso/issues/77483

## Proposed Changes

* Moved `NavigatorProvider` outside the `StepContainer` so we're able to get the navigator to navigate between screens
* Added the Activation notice to notify their homepage will be replaced
* Added the Activation screen, and clicking the “Continue” button will enter this screen
* Display the “Back” button on the Activation screen

**Screenshot**

<img width="300" src="https://github.com/Automattic/wp-calypso/assets/13596067/8f7e5da1-03f8-43ca-8f16-19ab293e8e0e" />

**Demo**

https://github.com/Automattic/wp-calypso/assets/13596067/6e1e05fc-d066-464e-b4b6-3653fefa9234

## Testing Instructions



* Use Calypso Live link [below](https://github.com/Automattic/wp-calypso/pull/77475#issuecomment-1566532503)
* Go to `/themes`
* Scroll down to select BCPA CTA
* On the Pattern Assembler screen
  * Select layout and styles
  * Continue
  * Ensure you're able to see the Activation screen
  * Click the “Back” button on the top-left corner, and ensure you're able to go back to the main screen
  * Continue again
  * Click the checkbox and the “Activate“ button
  * Ensure you're able to land on the site editor and see the selected layout and styles

## Pre-merge Checklist
